functests/performance/rt-kernel.go

```diff
@@ -0,0 +1,69 @@
+package performance
+
+import (
+	"fmt"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	testutils "github.com/openshift-kni/performance-addon-operators/functests/utils"
+	testclient "github.com/openshift-kni/performance-addon-operators/functests/utils/client"
+	"github.com/openshift-kni/performance-addon-operators/functests/utils/nodes"
+	"github.com/openshift-kni/performance-addon-operators/functests/utils/pods"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+var _ = Describe("[performance]RT Kernel", func() {
+	var workerRTNodes []corev1.Node
+
+	BeforeEach(func() {
+		var err error
+		workerRTNodes, err = nodes.GetByRole(testclient.Client, testutils.RoleWorkerRT)
+		Expect(err).ToNot(HaveOccurred())
+		Expect(workerRTNodes).ToNot(BeEmpty())
+	})
+
+	It("should have RT kernel enabled", func() {
+
+		Eventually(func() string {
+			var testpod *corev1.Pod
+
+			// run uname -a in a busybox pod and get logs
+			testpod = pods.GetBusybox()
+			testpod.Spec.Containers[0].Command = []string{"uname", "-a"}
+			testpod.Spec.RestartPolicy = corev1.RestartPolicyNever
+			testpod.Spec.NodeSelector = map[string]string{
+				fmt.Sprintf("%s/%s", testutils.LabelRole, testutils.RoleWorkerRT): "",
+			}
+
+			var err error
+			testpod, err = testclient.Client.Pods(testutils.NamespaceTesting).Create(testpod)
+			if err != nil {
+				return ""
+			}
+			defer func() {
```

Comment:
can we use an AfterEach() instead. I'm not 100% sure what happens if you do a `ctrl-c` to kill execution when a defer is in use. I know that ginkgo will attempt to run the AfterEach() functions even if you try to kill the test execution early though.

Reply:
fair point, will change it